### ISSUE TITLE
[Cloud target] The property 'IsControlAddIn' must be set on the .NET type (...)

### ISSUE THREAD

**Issue by JulienDecottignies** [microsoft/AL#5340]
Hi,

**Describe the bug**

When publishing extension from vscode or with powershell cmdlet to my container, it fails with this error message:
```
src/Pages/pages/Pag50100.MapPart.al(10,30): error AL0511: The property 'IsControlAddIn' must be set on the .NET type 'DocumentFormat.OpenXml.Spreadsheet.Map' if the type represents a .NET control add-in.
src/Pages/pages/Pag50100.MapPart.al(10,30): error AL0417: Control add-in 'Map' not found
```

Obviously, I can't create a dotnet reference to this assembly as I am targetting a Cloud version.

This piece of code worked in previous versionS, I just wanted to migrate it into the wave2 release.

Is there any declaration difference in this version of AL Language ?

**To Reproduce**

- Create a project with a controladdin
- Reference it in a Page
- Publish the app.

**Expected behavior**

App should be published

**Screenshots**

![image](https://user-images.githubusercontent.com/29598240/66464616-9dc21800-ea7f-11e9-8edf-75a7b593c2b0.png)

![image](https://user-images.githubusercontent.com/29598240/66464641-a87cad00-ea7f-11e9-9d60-5c1793c4c5b5.png)

**5. Versions:**
 - AL Language: 4.0.182565

 - Business Central: Version: Platform 15.0.36510.0 + Application 15.0.36560.0

Thanks for your help

**Comment by thpeder** [comment]
Hi @JulienDecottignies,

Thanks for reporting the issue, it seems like we are mapping the name of the ControlAddin to a .net type.
The workaround for now is to rename it so it doesn't use the same name.

**Comment by JulienDecottignies** [comment]
Yes that's it ! Thanks ! Renaming it with "MyMap" solved it!

**Comment by atoader** [comment]
Hi! I looked in the compiler code and the logic is as follows. The compiler searches for a .NET type that could be a .NET control add-in. If there is a type with that name, assume that the user intended to use it as a control add-in and verify that it has the IsControlAddIn property. If the type does not have the IsControlAddIn property set, report an error.  If no .NET type with that name can be found, the compiler searches for a native control add-in. 
This logic allows .NET add-ins to work side-by-side with native AL add-ins and replace them if needed while helping the user discover the correct syntax. This logic was added to support legacy constructs for applications migrating from C/AL to AL.
What is happening in your case is that a .NET Map type is declared in the base application and that is chosen over your native add-in.
Our recommendation is to use an prefix for your object names to prevent clashes like this one.
